runtime/vm/visible.c

```diff
@@ -142,3 +156,51 @@ checkVisibility(J9VMThread *currentThread, J9Class* sourceClass, J9Class* destCl
 
 	return result;
 }
+
+#if defined(J9VM_OPT_VALHALLA_NESTMATES)
+static UDATA loadAndVerifyNestHost(J9VMThread *vmThread, J9Class *clazz) {
+	J9Class *nestHost = NULL;
+	UDATA result = J9_VISIBILITY_ALLOWED;
+	J9ROMClass *romClass = clazz->romClass;
+	J9UTF8 *nestHostName = J9ROMCLASS_NESTHOSTNAME(romClass);
+
+	/* If no nest host is named, class is own nest host */
+	if (NULL == nestHostName) {
+		nestHost = clazz;
+	} else {
+		UDATA classLoadingFlags = J9_FINDCLASS_FLAG_THROW_ON_FAIL;
```

Comment:
TODO: check the code to see if we need to worry about this at JIT compile time.  If it's a compile time resolve, we hopefully aren't doing access already.

Reply:
This is called by `checkVisibility`, which is called by `resolveClassRef` during jit compile time resolves.  

This code needs to check if this is a compile time resolve and if it is, replace `J9_FINDCLASS_FLAG_THROW_ON_FAIL` with `J9_FINDCLASS_FLAG_EXISTING_ONLY`.

Reply:
`checkVisiblity` <- call by resolvefield.cpp code.  Needs to also respect `J9_RESOLVE_FLAG_NO_THROW_ON_FAIL` to indicate not to throw.

Reply:
```
if ((resolveFlags & J9_RESOLVE_FLAG_NO_THROW_ON_FAIL)
|| (resolveFlags & COMPILE_TIME)
|| .... AOT....
|| ... J9_FINDCLASS_FLAG_EXISTING_ONLY ...
) {
classLoadingFlags = J9_FINDCLASS_FLAG_EXISTING_ONLY;
}
```

Or make the callers pass in `BOOLEAN canRunJavaCode` and use that to pick the `EXISTING_ONLY` or not flags.  Some callers may already know this.